packages/mdc-typography/_mixins.scss

```diff
@@ -40,3 +40,35 @@
   white-space: nowrap;
   overflow: hidden;
 }
+
+@mixin mdc-typography-text-baseline($size, $trailing: false) {
+  $offset: ($size * 25) / 100;
+
+  line-height: normal;
+
+  &::before {
+    display: inline-block;
+    width: 0;
+    height: $size;
+    content: "";
+    vertical-align: 0;
+  }
+
+  @if $trailing {
```

Comment:
Okay I looked at #3085 and I think i understand (somewhat) why this mixin's logic is more complicated than I expected it to be.

I think that mdc-list has a strange HTML structure for supporting two line lists. I think if we change that HTML markup (so that it is not mixing text elements with span elements), then we can more simply implement the baseline alignment fix.